### PR TITLE
Feat/currency rates hook

### PR DESCRIPTION
**Pull request by baikov98**

### CODE REVIEW

File: packages/app/src/entities/currency-rates/api/currency-rates.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { CurrencyRates } from '../types'
+
+export const getCryptoCurrenciesRates = async () => {
+  try {
+    const result = await fetch('https://api.coinbase.com/v2/exchange-rates?currency=USD').then(
```
Comment (bouvens) [inline packages/app/src/entities/currency-rates/api/currency-rates.ts:5]: Let's do not mix `await` and `then` on one string.

It's also a good approach to extract `fetch` with its helper logic to a sort of `getRequest` function.

We already have `postData` in the landing project. We can rename it to `postRequest` and extract as well. This part is not needed right now.

File: packages/app/src/entities/currency-rates/context/provider.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    const apiRates = await getCryptoCurrenciesRates()
+    if (apiRates) setCurrencyRates(apiRates)
+  }
+
+  useEffect(() => {
+    getRates()
+    setInterval(() => {
+      getRates()
+    }, UPDATE_RATES_INTERVAL_MS)
+  }, [])
```
Comment (bouvens) [inline packages/app/src/entities/currency-rates/context/provider.tsx:23]: We have to clear interval on unmount.

```suggestion
  useEffect(() => {
    getRates()
    const intervalId = setInterval(() => {
      getRates()
    }, UPDATE_RATES_INTERVAL_MS)
    return () => {
      clearInterval(intervalId)
    }
  }, [])
```

**Review by bouvens — CHANGES_REQUESTED** [review]

File: packages/app/src/entities/currency-rates/context/provider.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const [currencyRates, setCurrencyRates] = useState<CurrencyRates>()
+
+  const getRates = async () => {
+    const apiRates = await getCryptoCurrenciesRates()
+    if (apiRates) setCurrencyRates(apiRates)
+  }
+
+  useEffect(() => {
+    getRates()
+    setInterval(() => {
```
[the remark this answers is not in the document]
Comment (baikov98) [reply packages/app/src/entities/currency-rates/context/provider.tsx]: added

File: packages/app/src/entities/currency-rates/context/provider.tsx (hunk 2 above)
Comment (baikov98) [reply packages/app/src/entities/currency-rates/context/provider.tsx]: added

File: packages/app/src/entities/currency-rates/api/currency-rates.ts (hunk 1 above)
Comment (bouvens) [reply packages/app/src/entities/currency-rates/api/currency-rates.ts]: Thank you

**Review by bouvens — APPROVED** [review]

**Pull request merged**